src/main/java/org/assertj/core/api/AbstractPathAssert.java

```diff
@@ -0,0 +1,821 @@
+ *     the tested path, assertions will throw a {@link
+ *     ProviderMismatchException}.</li>
+ *     <li>Some assertions may need to perform I/O on the path's underlying
+ *     filesystem; if an I/O error occurs when accessing the filesystem, these
+ *     assertions will throw a {@link PathsException}.</li>
+ * </ul>
+ *
+ * @param <S> self type
+ *
+ * @see Path
+ * @see java.nio.file.Paths#get(String, String...)
+ * @see FileSystem
+ * @see FileSystem#getPath(String, String...)
+ * @see FileSystems#getDefault()
+ * @see Files
+ */
+public abstract class AbstractPathAssert<S extends AbstractPathAssert<S>>
+    extends AbstractAssert<S, Path>
+{
+    @VisibleForTesting
+    protected Paths paths = Paths.instance();
+
+    protected AbstractPathAssert(final Path actual, final Class<?> selfType)
+    {
+        super(actual, selfType);
+    }
+
+    /**
+     * Assert that a given path exists
+     *
+     * <p><strong>Note that this assertion will follow symbolic links before
+     * asserting the path's existence.</strong></p>
+     *
+     * <p>On Windows system, this has no influence. On Unix systems, this means
+     * the assertion result will fail if the path is a symbolic link whose
+     * target does not exist. If you want to assert the existence of the
+     * symbolic link itself, use {@link #existsNoFollow()} instead.</p>
+     *
+     * <p>Examples:</p>
+     *
+     * <pre><code class="java">
+     * // fs is a Unix filesystem
+     * // Create a symbolic link whose target does not exist
+     * final Path nonExistent = fs.getPath("nonexistent");
+     * final Path dangling = fs.getPath("dangling");
+     * Files.createSymbolicLink(dangling, nonExistent);
+     *
+     * // Create a regular file, and a symbolic link pointing to it
+     * final Path someFile = fs.getPath("somefile");
+     * Files.createFile(someFile);
+     * final Path symlink = fs.getPath("symlink");
+     * Files.createSymbolicLink(symlink, someFile);
+     *
+     * // The following assertion succeeds
+     * assertThat(symlink).exists();
+     *
+     * // The following assertion fails
+     * assertThat(dangling).exists();
+     * </code></pre>
+     *
+     * @return self
+     *
+     * @see Files#exists(Path, LinkOption...)
+     */
+    public S exists()
+    {
+        paths.assertExists(info, actual);
+        return myself;
+    }
+
+    /**
+     * Assert that a given path exists, not following symbolic links
+     *
+     * <p>This assertion behaves like {@link #exists()}, with the difference
+     * that it can be used to assert the existence of a symbolic link even if
+     * its target is invalid.</p>
+     *
+     * <p>Examples:</p>
+     *
+     * <pre><code class="java">
+     * // fs is a Unix filesystem
+     * // Create a symbolic link whose target does not exist
+     * final Path nonExistent = fs.getPath("nonexistent");
+     * final Path dangling = fs.getPath("dangling");
```

Comment:
rename `dangling` to `linkToNonExistent` to make things more obvious (you almost don't have to read the code initializing `linkToNonExistent` to understand the assertions)